File: app/src/main/java/com/android/votechain/common/view/BaseFragment.java
package com.android.votechain.common.view;

import android.support.v4.app.Fragment;

/**
 * 27/11/15.
 * <p>
 * * <p>
 * A fragment like an activity only will execute operations that affect the UI.
 * These operations are defined by a view model and are triggered by its presenter.
 * </p>
 */

public abstract class BaseFragment extends Fragment {

  //@Override public View onCreateView(LayoutInflater inflater, ViewGroup container,
  //    Bundle savedInstanceState) {
  //  return inflater.inflate(getFragmentLayout(), container, false);
  //}
  //
  //@Override public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
  //  super.onViewCreated(view, savedInstanceState);
  //  setupActionBar(getSupportActionBar());
  //}
  //
  //protected void setupActionBar(ActionBar actionBar) {
  //
  //}
  //
  //protected void enableOptionsMenu() {
  //  setHasOptionsMenu(true);
  //}
  ///**
  // * Replace all the annotated fields with ButterKnife annotations with the proper value
  // */
  //
  //public void setResultAactivity(int resultCode, Intent data) {
  //  getActivity().setResult(resultCode, data);
  //}
  //
  //public void finishActivity() {
  //  getActivity().finish();
  //}
  //
  //public FragmentManager getActivitySupportFragmentManager() {
  //  return getActivity().getSupportFragmentManager();
  //}
  //
  //public ActionBar getSupportActionBar() {
  //  return ((BaseActivity) getActivity()).getSupportActionBar();
  //}
  //
  //public void addFragment(Fragment fragment) {
  //  ((BaseFragActivity) getActivity()).addFragment(fragment);
  //}
  //
  //public void addFragment(Fragment fragment, @AnimRes int enter, @AnimRes int exit,
  //    @AnimRes int popEnter, @AnimRes int popExit) {
  //  ((BaseFragActivity) getActivity()).addFragment(fragment, enter, exit, popEnter, popExit);
  //}
  //
  //public void replaceFragment(Fragment fragment) {
  //  ((BaseFragActivity) getActivity()).replaceFragment(fragment);
  //}
  //
  //public void replaceFragment(Fragment fragment, @AnimRes int enter, @AnimRes int exit) {
  //  ((BaseFragActivity) getActivity()).replaceFragment(fragment, enter, exit);
  //}
  //
  //public void popConcurrentFragment() {
  //  ((BaseFragActivity) getActivity()).popFragment();
  //}
  //
  //@Nullable public Toolbar getToolbar() {
  //  return ((BaseActivity) getActivity()).getToolbar();
  //}
  //
  ///**
  // * Specify the layout of the fragment to be inflated in the {@link BaseFragment#onCreateView(LayoutInflater,
  // * ViewGroup, Bundle)}
  // */
  //protected abstract int getFragmentLayout();
}

File: app/src/main/java/com/android/votechain/common/view/BaseFragment.kt
package com.android.votechain.common.view

import android.content.Intent
import android.os.Bundle
import android.support.annotation.AnimRes
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v7.app.ActionBar
import android.support.v7.widget.Toolbar
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.toolbar.*

/**
 * 27/11/15.
 *
 *
 * *
 *
 *
 * A fragment like an activity only will execute operations that affect the UI.
 * These operations are defined by a view model and are triggered by its presenter.
 *
 */

abstract class BaseFragment : Fragment() {


  override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
      savedInstanceState: Bundle?): View? {
    return inflater?.inflate(getFragmentLayout(), container, false)
  }

  abstract   fun getFragmentLayout(): Int

  override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)
    setupActionBar(getSupportActionBar())
  }

  private fun getSupportActionBar(): ActionBar? {
    return (activity as BaseActivity).supportActionBar
  }

  open fun setupActionBar(actionBar: ActionBar?) {

  }

  open fun enableOptionsMenu() {
    setHasOptionsMenu(true)
  }

  /**
   * Replace all the annotated fields with ButterKnife annotations with the proper value
   */

  fun setResultAactivity(resultCode: Int, data: Intent) {
    activity.setResult(resultCode, data)
  }

  fun finishActivity() {
    activity.finish()
  }

  fun addFragment(fragment: Fragment) {
    (activity as BaseFragActivity).addFragment(fragment)
  }

  fun addFragment(fragment: Fragment, @AnimRes enter: Int, @AnimRes exit: Int,
      @AnimRes popEnter: Int, @AnimRes popExit: Int) {
    (activity as BaseFragActivity).addFragment(fragment, enter, exit, popEnter, popExit)
  }

  fun replaceFragment(fragment: Fragment) {
    (activity as BaseFragActivity).replaceFragment(fragment)
  }

  fun replaceFragment(fragment: Fragment, @AnimRes enter: Int, @AnimRes exit: Int) {
    (activity as BaseFragActivity).replaceFragment(fragment, enter, exit)
  }

  fun popConcurrentFragment() {
    (activity as BaseFragActivity).popFragment()
  }

   fun getToolbar() : Toolbar? {
    return (activity as BaseActivity).toolbar;
  }

}

